backends/p4tools/modules/testgen/core/target.h

```diff
@@ -64,7 +61,10 @@ class TestgenTarget : public Target {
     virtual ExprStepper *getExprStepperImpl(ExecutionState &state, AbstractSolver &solver,
                                             const ProgramInfo &programInfo) const = 0;
 
-    explicit TestgenTarget(std::string deviceName, std::string archName);
+    explicit TestgenTarget(const std::string &deviceName, const std::string &archName);
+
+ private:
+    CompilerResultOrError runCompilerImpl(const IR::P4Program *program) const override;
```

Comment:
Should this be `private` if it is a virtual function? Maybe make it `protected` so that it can be called from a derived class.